diff --git a/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt b/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
--- a/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
+++ b/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
@@ -26,7 +26,7 @@
 ; DO NOT CHANGE R2R NUMERIC VALUES OF THE EXISTING SETS. Changing R2R numeric values definitions would be R2R format breaking change.
 
 ; The ISA definitions should also be mapped to `hwintrinsicIsaRangeArray` in hwintrinsic.cpp.
-; NEXT_AVAILABLE_R2R_BIT = 94
+; NEXT_AVAILABLE_R2R_BIT = 99
 
 ; Definition of X86 instruction sets
 definearch         ,X86   ,32Bit                ,X64, X64, X86
@@ -284,11 +284,19 @@ instructionset     ,RiscV64 , ,Zba               ,57 ,Zba                 ,zba
 instructionset     ,RiscV64 , ,Zbb               ,58 ,Zbb                 ,zbb
 instructionset     ,RiscV64 , ,Zbs               ,84 ,Zbs                 ,zbs
 instructionset     ,RiscV64 , ,Zicond            ,94 ,Zicond              ,zicond
+instructionset     ,RiscV64 , ,F                 ,95 ,F                   ,f
+instructionset     ,RiscV64 , ,D                 ,96 ,D                   ,d
+instructionset     ,RiscV64 , ,C                 ,97 ,C                   ,c
+instructionset     ,RiscV64 , ,A                 ,98 ,A                   ,a
 
 implication        ,RiscV64 ,Zbb                 ,RiscV64Base
 implication        ,RiscV64 ,Zba                 ,RiscV64Base
 implication        ,RiscV64 ,Zbs                 ,RiscV64Base
 implication        ,RiscV64 ,Zicond              ,RiscV64Base
+implication        ,RiscV64 ,F                   ,RiscV64Base
+implication        ,RiscV64 ,D                   ,F
+implication        ,RiscV64 ,C                   ,RiscV64Base
+implication        ,RiscV64 ,A                   ,RiscV64Base
 
 ;                    ,name and aliases           ,archs    ,lower baselines included by implication
 instructionsetgroup  ,x86-64-v2                  ,X64 X86  ,base
diff --git a/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSet.cs b/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSet.cs
--- a/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSet.cs
+++ b/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSet.cs
@@ -105,5 +105,9 @@ namespace Internal.ReadyToRunConstants
         PackedSimd = 92,
         Cssc = 93,
         Zicond = 94,
+        F = 95,
+        D = 96,
+        C = 97,
+        A = 98,
     }
 }
diff --git a/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSetHelper.cs b/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSetHelper.cs
--- a/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSetHelper.cs
+++ b/src/coreclr/tools/Common/Internal/Runtime/ReadyToRunInstructionSetHelper.cs
@@ -75,6 +75,10 @@ namespace Internal.ReadyToRunConstants
                         case InstructionSet.RiscV64_Zbb: return ReadyToRunInstructionSet.Zbb;
                         case InstructionSet.RiscV64_Zbs: return ReadyToRunInstructionSet.Zbs;
                         case InstructionSet.RiscV64_Zicond: return ReadyToRunInstructionSet.Zicond;
+                        case InstructionSet.RiscV64_F: return ReadyToRunInstructionSet.F;
+                        case InstructionSet.RiscV64_D: return ReadyToRunInstructionSet.D;
+                        case InstructionSet.RiscV64_C: return ReadyToRunInstructionSet.C;
+                        case InstructionSet.RiscV64_A: return ReadyToRunInstructionSet.A;
 
                         default: throw new Exception("Unknown instruction set");
                     }
diff --git a/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs b/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs
--- a/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs
+++ b/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs
@@ -61,6 +61,10 @@ namespace Internal.JitInterface
         RiscV64_Zbb = InstructionSet_RiscV64.Zbb,
         RiscV64_Zbs = InstructionSet_RiscV64.Zbs,
         RiscV64_Zicond = InstructionSet_RiscV64.Zicond,
+        RiscV64_F = InstructionSet_RiscV64.F,
+        RiscV64_D = InstructionSet_RiscV64.D,
+        RiscV64_C = InstructionSet_RiscV64.C,
+        RiscV64_A = InstructionSet_RiscV64.A,
         Wasm32_WasmBase = InstructionSet_Wasm32.WasmBase,
         Wasm32_PackedSimd = InstructionSet_Wasm32.PackedSimd,
         Wasm32_Vector128 = InstructionSet_Wasm32.Vector128,
@@ -211,6 +215,10 @@ namespace Internal.JitInterface
         Zbb = 3,
         Zbs = 4,
         Zicond = 5,
+        F = 6,
+        D = 7,
+        C = 8,
+        A = 9,
     }
 
     public enum InstructionSet_Wasm32
@@ -605,6 +613,14 @@ namespace Internal.JitInterface
                             resultflags.AddInstructionSet(InstructionSet.RiscV64_RiscV64Base);
                         if (resultflags.HasInstructionSet(InstructionSet.RiscV64_Zicond))
                             resultflags.AddInstructionSet(InstructionSet.RiscV64_RiscV64Base);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_F))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_RiscV64Base);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_D))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_F);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_C))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_RiscV64Base);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_A))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_RiscV64Base);
                         break;
 
                     case TargetArchitecture.Wasm32:
@@ -912,6 +928,14 @@ namespace Internal.JitInterface
                             resultflags.AddInstructionSet(InstructionSet.RiscV64_Zbs);
                         if (resultflags.HasInstructionSet(InstructionSet.RiscV64_RiscV64Base))
                             resultflags.AddInstructionSet(InstructionSet.RiscV64_Zicond);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_RiscV64Base))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_F);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_F))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_D);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_RiscV64Base))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_C);
+                        if (resultflags.HasInstructionSet(InstructionSet.RiscV64_RiscV64Base))
+                            resultflags.AddInstructionSet(InstructionSet.RiscV64_A);
                         break;
 
                     case TargetArchitecture.Wasm32:
@@ -1173,6 +1197,10 @@ namespace Internal.JitInterface
                     yield return new InstructionSetInfo("zbb", "", InstructionSet.RiscV64_Zbb, true);
                     yield return new InstructionSetInfo("zbs", "", InstructionSet.RiscV64_Zbs, true);
                     yield return new InstructionSetInfo("zicond", "", InstructionSet.RiscV64_Zicond, true);
+                    yield return new InstructionSetInfo("f", "", InstructionSet.RiscV64_F, true);
+                    yield return new InstructionSetInfo("d", "", InstructionSet.RiscV64_D, true);
+                    yield return new InstructionSetInfo("c", "", InstructionSet.RiscV64_C, true);
+                    yield return new InstructionSetInfo("a", "", InstructionSet.RiscV64_A, true);
                     break;
 
                 case TargetArchitecture.Wasm32:
diff --git a/src/coreclr/inc/corinfoinstructionset.h b/src/coreclr/inc/corinfoinstructionset.h
--- a/src/coreclr/inc/corinfoinstructionset.h
+++ b/src/coreclr/inc/corinfoinstructionset.h
@@ -63,6 +63,10 @@ enum CORINFO_InstructionSet
     InstructionSet_Zbb=3,
     InstructionSet_Zbs=4,
     InstructionSet_Zicond=5,
+    InstructionSet_F=6,
+    InstructionSet_D=7,
+    InstructionSet_C=8,
+    InstructionSet_A=9,
 #endif // TARGET_RISCV64
 #ifdef TARGET_WASM
     InstructionSet_WasmBase=1,
@@ -459,6 +463,14 @@ inline CORINFO_InstructionSetFlags EnsureInstructionSetFlagsAreValid(CORINFO_Ins
             resultflags.RemoveInstructionSet(InstructionSet_Zbs);
         if (resultflags.HasInstructionSet(InstructionSet_Zicond) && !resultflags.HasInstructionSet(InstructionSet_RiscV64Base))
             resultflags.RemoveInstructionSet(InstructionSet_Zicond);
+        if (resultflags.HasInstructionSet(InstructionSet_F) && !resultflags.HasInstructionSet(InstructionSet_RiscV64Base))
+            resultflags.RemoveInstructionSet(InstructionSet_F);
+        if (resultflags.HasInstructionSet(InstructionSet_D) && !resultflags.HasInstructionSet(InstructionSet_F))
+            resultflags.RemoveInstructionSet(InstructionSet_D);
+        if (resultflags.HasInstructionSet(InstructionSet_C) && !resultflags.HasInstructionSet(InstructionSet_RiscV64Base))
+            resultflags.RemoveInstructionSet(InstructionSet_C);
+        if (resultflags.HasInstructionSet(InstructionSet_A) && !resultflags.HasInstructionSet(InstructionSet_RiscV64Base))
+            resultflags.RemoveInstructionSet(InstructionSet_A);
 #endif // TARGET_RISCV64
 #ifdef TARGET_WASM
         if (resultflags.HasInstructionSet(InstructionSet_Vector128) && !resultflags.HasInstructionSet(InstructionSet_PackedSimd))
@@ -763,6 +775,14 @@ inline const char *InstructionSetToString(CORINFO_InstructionSet instructionSet)
             return "Zbs";
         case InstructionSet_Zicond :
             return "Zicond";
+        case InstructionSet_F :
+            return "F";
+        case InstructionSet_D :
+            return "D";
+        case InstructionSet_C :
+            return "C";
+        case InstructionSet_A :
+            return "A";
 #endif // TARGET_RISCV64
 #ifdef TARGET_WASM
         case InstructionSet_WasmBase :
@@ -974,6 +994,10 @@ inline CORINFO_InstructionSet InstructionSetFromR2RInstructionSet(ReadyToRunInst
         case READYTORUN_INSTRUCTION_Zbb: return InstructionSet_Zbb;
         case READYTORUN_INSTRUCTION_Zbs: return InstructionSet_Zbs;
         case READYTORUN_INSTRUCTION_Zicond: return InstructionSet_Zicond;
+        case READYTORUN_INSTRUCTION_F: return InstructionSet_F;
+        case READYTORUN_INSTRUCTION_D: return InstructionSet_D;
+        case READYTORUN_INSTRUCTION_C: return InstructionSet_C;
+        case READYTORUN_INSTRUCTION_A: return InstructionSet_A;
 #endif // TARGET_RISCV64
 #ifdef TARGET_WASM
         case READYTORUN_INSTRUCTION_WasmBase: return InstructionSet_WasmBase;
diff --git a/src/coreclr/inc/readytoruninstructionset.h b/src/coreclr/inc/readytoruninstructionset.h
--- a/src/coreclr/inc/readytoruninstructionset.h
+++ b/src/coreclr/inc/readytoruninstructionset.h
@@ -102,6 +102,11 @@ enum ReadyToRunInstructionSet
     READYTORUN_INSTRUCTION_PackedSimd=92,
     READYTORUN_INSTRUCTION_Cssc=93,
     READYTORUN_INSTRUCTION_Zicond=94,
+    READYTORUN_INSTRUCTION_F=95,
+    READYTORUN_INSTRUCTION_D=96,
+    READYTORUN_INSTRUCTION_C=97,
+    READYTORUN_INSTRUCTION_A=98,
+
 };
 
 #endif // READYTORUNINSTRUCTIONSET_H
diff --git a/src/coreclr/inc/jiteeversionguid.h b/src/coreclr/inc/jiteeversionguid.h
--- a/src/coreclr/inc/jiteeversionguid.h
+++ b/src/coreclr/inc/jiteeversionguid.h
@@ -37,11 +37,11 @@
 
 #include <minipal/guid.h>
 
-constexpr GUID JITEEVersionIdentifier = { /* 65743063-e8fa-41d4-9496-c436974c00f5 */
-    0x65743063,
-    0xe8fa,
-    0x41d4,
-    {0x94, 0x96, 0xc4, 0x36, 0x97, 0x4c, 0x00, 0xf5}
+constexpr GUID JITEEVersionIdentifier = { /* 2348bb76-27d3-449f-824f-1e51c888a151 */
+    0x2348bb76,
+    0x27d3,
+    0x449f,
+    {0x82, 0x4f, 0x1e, 0x51, 0xc8, 0x88, 0xa1, 0x51}
   };
 
 #endif // JIT_EE_VERSIONING_GUID_H
