diff --git a/eng/native/configurecompiler.cmake b/eng/native/configurecompiler.cmake
index e8ad8615f6d..2020742f4a0 100644
--- a/eng/native/configurecompiler.cmake
+++ b/eng/native/configurecompiler.cmake
@@ -782,6 +782,11 @@ if(CLR_CMAKE_HOST_UNIX_X86)
   add_compile_options(-msse2)
 endif()
 
+if(CLR_CMAKE_HOST_UNIX_RISCV64)
+  # Remove compressed instructions and D extension
+  add_compile_options(-march=rv64ima)
+endif()
+
 if(CLR_CMAKE_HOST_UNIX)
   add_compile_options(${CLR_ADDITIONAL_COMPILER_OPTIONS})
 endif(CLR_CMAKE_HOST_UNIX)
