diff --git a/eng/common/cross/build-rootfs.sh b/eng/common/cross/build-rootfs.sh
--- a/eng/common/cross/build-rootfs.sh
+++ b/eng/common/cross/build-rootfs.sh
@@ -42,8 +42,8 @@
 __AlpinePackages="alpine-base"
 __AlpinePackages+=" build-base"
 __AlpinePackages+=" linux-headers"
-__AlpinePackages+=" lldb-dev"
-__AlpinePackages+=" python3"
+#__AlpinePackages+=" lldb-dev"
+#__AlpinePackages+=" python3"
 __AlpinePackages+=" libedit"
 
 # symlinks fixer
@@ -57,8 +57,8 @@
 __AlpinePackages+=" gettext-dev"
 __AlpinePackages+=" icu-dev"
 __AlpinePackages+=" libunwind-dev"
-__AlpinePackages+=" lttng-ust-dev"
-__AlpinePackages+=" compiler-rt"
+#__AlpinePackages+=" lttng-ust-dev"
+#__AlpinePackages+=" compiler-rt"
 
 # runtime libraries' dependencies
 __UbuntuPackages+=" libcurl4-openssl-dev"
@@ -68,7 +68,7 @@
 __UbuntuPackages+=" libbrotli-dev"
 
 __AlpinePackages+=" curl-dev"
-__AlpinePackages+=" krb5-dev"
+__AlpinePackages+=" "
 __AlpinePackages+=" openssl-dev"
 __AlpinePackages+=" zlib-dev"
 
@@ -132,7 +132,7 @@
 '
 __Keyring=
 __KeyringFile="/usr/share/keyrings/ubuntu-archive-keyring.gpg"
-__SkipSigCheck=0
+__SkipSigCheck=1
 __SkipEmulation=0
 __UseMirror=0
 
@@ -532,7 +532,7 @@
     done
 
     if [[ "$__SkipSigCheck" == "1" ]]; then
-        __ApkSignatureArg="--allow-untrusted"
+        __ApkSignatureArg="--allow-untrusted --no-check-certificate"
     else
         __ApkSignatureArg="--keys-dir $__ApkKeysDir"
     fi
